server/schedulers/hot_region.go

```diff
@@ -563,10 +569,8 @@ func (bs *balanceSolver) checkSrcByDimPriorityAndTolerance(minLoad, expectLoad *
 
 // filterHotPeers filtered hot peers from statistics.HotPeerStat and deleted the peer if its region is in pending status.
 // The returned hotPeer count in controlled by `max-peer-number`.
-func (bs *balanceSolver) filterHotPeers() []*statistics.HotPeerStat {
-	ret := bs.cur.srcStore.HotPeers
-	// Return at most MaxPeerNum peers, to prevent balanceSolver.solve() too slow.
-	maxPeerNum := bs.sche.conf.GetMaxPeerNumber()
+func (bs *balanceSolver) filterHotPeers(storeLoad *statistics.StoreLoadDetail) []*statistics.HotPeerStat {
+	ret := storeLoad.HotPeers
 
 	// filter pending region
 	appendItem := func(items []*statistics.HotPeerStat, item *statistics.HotPeerStat) []*statistics.HotPeerStat {
```

Comment:
how about refactor to `func(item *statistics.HotPeerStat)[]*statistics.HotPeerStat`?